backend/__tests__/posts.js

```diff
@@ -402,4 +402,146 @@ describe('Posts', () => {
+            it('should return response code of 201', async () => {
+                const user1 = await models.users.create({
+                    username: 'gandalf',
+                    nickname: 'gandalf',
+                    email: 'gandalf@gandalf.com',
+                    password: 'password',
+                })
+
+                const authToken = Authentication.generateAuthToken(user1)
+
+                const newPost = await models.posts.create({
+                    text_content: 'Test text',
+                    author: user1.id,
+                    parent: null,
+                })
+
+                const response = await request(server)
+                    .post(`/api/posts/${newPost.id}/like`)
+                    .set('Authorization', `Bearer ${authToken}`)
+
+                expect(response.statusCode).toBe(201)
+            })
+        })
+
+        describe('when user like others post in a valid way', () => {
+            it('should return response code of 201', async () => {
+                const user1 = await models.users.create({
+                    username: 'gandalf',
+                    nickname: 'gandalf',
+                    email: 'gandalf@gandalf.com',
+                    password: 'password',
+                })
+
+                const user2 = await models.users.create({
+                    username: 'gandalf2',
+                    nickname: 'gandalf2',
+                    email: 'gandalf2@gandalf.com',
+                    password: 'password2',
+                })
+
+                const authToken = Authentication.generateAuthToken(user2)
+
+                const newPost = await models.posts.create({
+                    text_content: 'Test text',
+                    author: user1.id,
+                    parent: null,
+                })
+
+                const response = await request(server)
+                    .post(`/api/posts/${newPost.id}/like`)
+                    .set('Authorization', `Bearer ${authToken}`)
+
+                expect(response.statusCode).toBe(201)
+            })
+        })
+
+        describe('when user try to like post that does not exist', () => {
+            it('should return response code of 404', async () => {
+                const user1 = await models.users.create({
+                    username: 'gandalf',
+                    nickname: 'gandalf',
+                    email: 'gandalf@gandalf.com',
+                    password: 'password',
+                })
+
+                const authToken = Authentication.generateAuthToken(user1)
+
+                const newPost = await models.posts.create({
+                    text_content: 'Test text',
+                    author: user1.id,
+                    parent: null,
+                })
+
+                const invalidId = newPost.id + 999
+
+                const response = await request(server)
+                    .put(`/api/posts/${invalidId}/like`)
```

Comment:
I think this part is supposed to be post instead of put.

Reply:
Yes. you are completely right!.
Sorry about that.